sale_channel_hook_stock_variation/models/stock_move.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2021 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _name = "stock.move"
+    _inherit = ["stock.move", "sale.channel.hook.mixin"]
+
+    def _notify_stock_variation(self):
+        for rec in self:
+            products_moved = rec.move_line_ids.mapped("product_id")
+            warehouse = rec.warehouse_id
```

Comment:
wrong logic.

Reply:
waht we should do:
for each product => create a job (with identity_key option to avoid recreating existing job https://github.com/OCA/queue/blob/13.0/queue_job/job.py#L103)

In each job for each binding we look at the stock level (in the right warehouse of the channel) and check the last value sent, if the value is not the same we recompute and sync the data)

Reply:
the hook mechanism already takes care of creating a job. This means we would have 2 jobs for each stock variation: one for checking the last value sent, one for sending the notification to the other webservice right ?

Reply:
identity_key: if I understand correctly, only the oldest job (so, the oldest update) will be kept, so if the job queue is long, then a newer update might get skipped right ?